src/main/java/interface_adapters/Controller.java

```diff
@@ -0,0 +1,97 @@
+package interface_adapters;
+
+import game_data.GameInstance;
+import usecase_gamedata.MapFactory;
+import usecase_gamedata.PlayerFactory;
+import usecase_playeractions.ActionManager;
+import usecase_playeractions.Map;
+
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+
+/**
+ * The Controller used to control player.
+ */
+public class Controller implements KeyListener {
+
+    /**
+     * Contain actions that depend on a Map.
+     * Don't have to be a Mover.
+     */
+    private ActionManager moveManager;
+
+    /**
+     * Contain actions that depend on a Player only.
+     */
+    private ActionManager manager;
+    private PlayerFactory playerFactory;
+
+
+    /**
+     * The default keys:
+     * 0:Heal
+     * 1:Heal-yes
+     * 2:Heal-No
+     * 3:Upgrade
+     * 4:Upgrade-Armor
+     * 5:Upgrade-Sword
+     * 6:Fight
+     * 7:Flee
+     * 8,9,10,11:moveUP,moveLeft,moveDown,moveRight
+     */
+    private final String[] KEYS = new String[]{"H","Y","N","U","A","S","F","R","W","A","S","D"};
```

Comment:
I might not mention it clearly enough in the TUT, but you would better change it through your own branch. I just want to make every pull request perfect enough.
Heal-yes and Upgrade should be deleted and Heal-no should change to leaveUpgradeHealMode. Upgrade Upgrade-Armor and Upgrade-Weapon's keypress will become"2","1"
private final String[] KEYS = new String[]{"H","N","2","1","F","R","W","A","S","D"};

Reply:
we need to add some keys.
     * 0:Heal(H)
     * 1:yes(Y)
     * 2:No(N)
     * 3:leaveHealAndUpgrade(L)
     * 4:Upgrade-Sword(1)
     * 5:Upgrade-Armor(2)
     * 6:Fight(F)
     * 7:Flee/run(R)
     * 8,9,10,11:moveUP,moveLeft,moveDown,moveRight
     * 12 skip(space)
     * private final String[] KEYS = new String[]{"H","Y","N","1","2","F","R","W","A","S","D"," "};